### ISSUE TITLE
Relative imports in a submodule are unresolved, but absolute ones work

### ISSUE THREAD

**Issue by ipkiss42** [microsoft/pyright#10948]
**Describe the bug**
Extract the attached code, go to the `repro` directory and run `pyright .`. It ouputs the following errors:

```sh
> pyright .
/tmp/repro/src/foo/__init__.py
  /tmp/repro/src/foo/__init__.py:1:21 - error: "Baz" is unknown import symbol (reportAttributeAccessIssue)
/tmp/repro/src/foo/bar/__init__.py
  /tmp/repro/src/foo/bar/__init__.py:2:6 - error: Import ".baz" could not be resolved (reportMissingImports)
/tmp/repro/src/main.py
  /tmp/repro/src/main.py:1:17 - error: "Baz" is unknown import symbol (reportAttributeAccessIssue)
  /tmp/repro/src/main.py:2:21 - error: "Baz" is unknown import symbol (reportAttributeAccessIssue)
4 errors, 0 warnings, 0 informations
```

The imports are correct, because `python src/main.py` works.

Also note that if you replace the relative import with an absolute one (see commented line in `src/foo/bar/__init__.py`) Pyright does not find errors anymore.

**Code or Screenshots**
Attached.

**VS Code extension or command-line**
Using Pyright 1.1.405 from the command-line

**Comment by ipkiss42** [comment]
My file upload failed, but anyway I realized that the issue is related to symlinks. I'm resolving this issue until I find a better way to reproduce it.
